kong/db/dao/init.lua

```diff
@@ -431,26 +431,40 @@ local function check_update(self, key, entity, options, name)
 end
 
 
-local function check_upsert(self, entity, options, name, value)
-  local entity_to_upsert, err = self.schema:process_auto_fields(entity, "upsert")
+local function check_upsert(self, key, entity, options, name)
```

Comment:
why did `value` change to `key`?

Reply:
Key makes more sense. It is either `unique value` used to select `by unique key` or the `primary key`.